lib/extensions/ar_adapter/ar_dba/postgresql.rb

```diff
@@ -874,4 +874,37 @@ def vacuum_analyze_table(table)
   def vacuum_full_analyze_table(table)
     execute("VACUUM FULL ANALYZE #{quote_table_name(table)}")
   end
+
+  def self.parse_dsn(dsn)
+    scanner = StringScanner.new(dsn)
+
+    dsn_hash = {}
+
+    until scanner.eos?
+      # get the key by matching up to and including the (optionally) white space bordered '='
+      key = scanner.scan_until(/\s?=\s?/)
+      key = key[0...-scanner.matched_size]
+
+      dsn_hash[key.to_sym] = get_dsn_value(scanner)
```

Comment:
I would leave this method as just a string parser, I think...the caller can do the int and password removal.